wa/instruments/energy_measurement.py

```diff
@@ -536,5 +538,9 @@ def extract_metrics(self, context):
                 context.add_metric(name, value, units)
 
     def teardown(self, context):
+        logger.debug('Job status {}'.format(context.current_job.status))
+        if context.current_job.status == 'FAILED':
+            self.update_output(context)
```

Comment:
Output should not be updated during teardown stage. If this really needs to be done, then you can register a callback for JOB_OUTPUT_PROCESSED signal. However, as @marcbonnici  commented already -- why is this being done? Generally, if a job has failed than any data generated for it is suspect, and should not be used. If this is a hack to workaround a specific issue then it should not be upstreamed.

Reply:
The usecase is mainly for a methodology where the an iteration is run back to back for a period of time, without the setup and teardown. However, if a run fails due to any reason after a certain time, then the collected results go to __failed directory for the workload,. But for the energy instrument, the energr_measurement.csv file is reset and all the data till that point is lost. 
This change will put the data in the __failed directory for whatever has been collected. It will be on the user to device if it should be kept of discarded.

Reply:
My point remains-- if the job has failed, the data should be discarded. You're dealing with expected intermittent failures in the underlying application, then it should be handled at the workload level, so the job would not be marked as failed. If the failures are not "expected" and are a result of unstable platform, etc, then registering a handler for JOB_FAILED signal (contingent on user configuring a parameter, i.e. not as the default) would be a better solution (though still feels a bit hacky).

Reply:
ok thank you.